server/ingest/handler/ingest_handler.go

```diff
@@ -117,63 +116,31 @@ func (h *IngestHandler) IndexContent(ctx context.Context, data []byte) error {
 
 const maxAnnounceSize = 512
 
-type announceMessage dtsync.Message
-
-// custom unmarshal because multiaddr.Multiaddr doesn't natively support json.Unmarshal
-func (a *announceMessage) UnmarshalJSON(data []byte) error {
-	top := map[string]*json.RawMessage{}
-	if err := json.Unmarshal(data, &top); err != nil {
-		return err
-	}
-	fmt.Printf("top: %+v\n", top)
-	ci, ok := top["Cid"]
-	if !ok || ci == nil {
-		return fmt.Errorf("missing cid")
-	}
-	c := cid.Cid{}
-	if err := json.Unmarshal(*ci, &c); err != nil {
+func (h *IngestHandler) Announce(ctx context.Context, r io.Reader) error {
+	data, err := io.ReadAll(io.LimitReader(r, maxAnnounceSize))
+	if err != nil {
 		return err
 	}
-	a.Cid = c
 
-	addrs, ok := top["Addrs"]
-	if !ok {
-		return fmt.Errorf("missing addrs")
-	}
-	addrList := make([]*json.RawMessage, 0)
-	if err := json.Unmarshal(*addrs, &addrList); err != nil {
+	// Decode CID and originator addresses from message.
+	an := dtsync.Message{}
+	if err = an.UnmarshalCBOR(bytes.NewBuffer(data)); err != nil {
```

Comment:
nit, we can use bytes.NewReader since we don't actually need a buffer.

Reply:
Also we don't need to call io.ReadAll above, we can just pass the limit reader in here.